test/reduce.jl

```diff
@@ -520,6 +520,12 @@ struct NonFunctionIsZero end
 @test count(NonFunctionIsZero(), [0]) == 1
 @test count(NonFunctionIsZero(), [1]) == 0
 
+@test count(Iterators.repeated(true, 3), init=0x00) isa UInt8
+@test count(!=(2), Iterators.take(1:7, 3), init=Int32(0)) isa Int32
+@test count(identity, [true, false], init=Int8(0)) isa Int8
+@test count(!, [true false; false true], dims=:, init=Int16(0)) isa Int16
+@test count(identity, [true false; false true], dims=2, init=UInt(0)) isa Matrix{UInt}
```

Comment:
Would be good to add tests with nonzero `init`, and also which actually test the return value (vs. only the type).